### PR TITLE
fix(medusa): Allow `method.data` to be passed when creating/updating ShippingMethods in ClaimService

### PR DESCRIPTION
**Pull request by kasperkristensen**
**What**
- Allows passing data on shipping methods during claim creation and updates. Defaults to an empty object.

**Testing** 
- Updates a test so it also passes along shipping method data.

### CODE REVIEW

File: packages/medusa/src/services/claim.ts [hunk 1]
```diff
@@ -166,7 +166,7 @@ export default class ClaimService extends TransactionBaseService {
             } else {
               await shippingOptionServiceTx.createShippingMethod(
                 method.option_id as string,
-                (method as any).data,
+                {},
```
Comment (olivermrbl) [inline packages/medusa/src/services/claim.ts:169]: question: Should we allow data on updates?

Could be: 
```
(method as any)?.data ?? {}
```

Comment (olivermrbl) [reply packages/medusa/src/services/claim.ts:169]: Ah wait. I see that we do not allow the `data` prop at all in the API layer. Fine to keep PR as is, and then add later. But if you think adding `data` to update + create is a minor addition, we might as well go ahead with it and do:
```
(method as any)?.data ?? {}
```

...in both update and create claim.

**Review by olivermrbl — APPROVED** [review]
LGTM w. comments :)

File: packages/medusa/src/services/claim.ts (hunk 1 above)
Comment (kasperkristensen) [reply packages/medusa/src/services/claim.ts:169]: I can add it real quick, but do we also want to add it for returns while we are at it? @olivermrbl

**Pull request merged**